torchrl/collectors/collectors.py

```diff
@@ -123,27 +123,45 @@ def _get_policy_and_device(
                 raise ValueError(
                     "env must be provided to _get_policy_and_device if policy is None"
                 )
-            policy = RandomPolicy(self.env.action_spec)
+            policy = RandomPolicy(self.env.action_spec)  # Get policy from the env
+
         try:
-            policy_device = next(policy.parameters()).device
+            policy_device = next(
+                policy.parameters()
+            ).device  # Get the first device from the policy parameter list
         except:  # noqa
             policy_device = (
-                torch.device(device) if device is not None else torch.device("cpu")
+                torch.device(device)
```

Comment:
I believe nothing has changed here either, let's remove the comment and set it back to the original